cli_docs/generate_cli_docs.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+import sys
+
+import click
+from pkg_resources import iter_entry_points
+
+
+def generate_docs(name, filename):
+    console_scripts = [ep for ep in iter_entry_points('console_scripts', name=name)]
+
+    if len(console_scripts) == 0:
+        raise click.ClickException('"{}" is not an installed console script.'.format(name))
+
+    # read all lines from original file
+    original_lines = []
+    with open(filename, "r") as md_file:
+        original_lines = md_file.readlines()
+
+    with open(filename, "w") as md_file:
+        # copy header from original file
+        for line in original_lines:
+            md_file.write(line)
+            if "AUTOGEN_START" in line:
+                break
+
+        # generate docs for cli
+        for entry_point in console_scripts:
+            cli = entry_point.resolve()
+            generate_command_docs(name, cli, md_file)
+
+
+def generate_command_docs(name, command, md_file, parent_ctx=None):
+    ctx = click.Context(command, info_name=name, parent=parent_ctx)
+    sub_commands = getattr(command, "commands", {})
+
+    # generate docs only for actual commands (not command groups)
+    if len(sub_commands) == 0:
+        cmd_path = ctx.command_path.split()
+        cmd_path.pop(0)
+        md_file.write("#### {}\n".format(" ".join(cmd_path)))
+        md_file.write("```\n{}\n```\n\n".format(ctx.get_help()))
+    else:
+        # if command has sub-commands - recursively generate docs for all sub-commands
+        for sub_cmd_name, sub_command in sub_commands.items():
+            generate_command_docs(sub_cmd_name, sub_command, md_file, ctx)
+
+
+if __name__ == '__main__':
+    print("Generating docs for bubuku-cli (make sure you installed the latest version of bubuku on a system)...")
+    generate_docs("bubuku-cli", "cli.md")
+    print("Done (updated 'cli.md' with latest docs)")
```

Comment:
You know, I have one problem with it. The purpose of first print was to show the user some message before script starts doing something, as the actual generation can take few seconds - so that user knows that work is in progress. With `print` it worked fine. When I switched to logging - it outputs both messages after the script finished, which is very disappointing.

Reply:
Ok, I checked again - it is not caused by logging but by something else. So, ignore my previous comment.

Reply:
But there is actually another problem. I don't like this "INFO:" before the messages it outputs. It looks like some log file. I just want to give a short hint to the user.

Reply:
And in general for me it's not exactly clear why to use stderr here instead of stdout. Why these messages can't be output of my script and should be treated as some error/diagnostics information.